addons/account/models/account_bank_statement.py

```diff
@@ -170,303 +45,191 @@ def _get_move_line_count(self):
-    date = fields.Date(required=True, states={'confirm': [('readonly', True)]}, index=True, copy=False, default=fields.Date.context_today)
-    date_done = fields.Datetime(string="Closed On")
-    balance_start = fields.Monetary(string='Starting Balance', states={'confirm': [('readonly', True)]}, compute='_compute_starting_balance', readonly=False, store=True, tracking=True)
-    balance_end_real = fields.Monetary('Ending Balance', states={'confirm': [('readonly', True)]}, compute='_compute_ending_balance', recursive=True, readonly=False, store=True, tracking=True)
-    state = fields.Selection(string='Status', required=True, readonly=True, copy=False, tracking=True, selection=[
-            ('open', 'New'),
-            ('posted', 'Processing'),
-            ('confirm', 'Validated'),
-        ], default='open',
-        help="The current state of your bank statement:"
-             "- New: Fully editable with draft Journal Entries."
-             "- Processing: No longer editable with posted Journal entries, ready for the reconciliation."
-             "- Validated: All lines are reconciled. There is nothing left to process.")
+    @api.depends('line_ids', 'line_ids.date')
+    def _compute_date(self):
+        for statement in self:
+            statement.date = bool(statement.line_ids) and max(statement.line_ids.mapped('date'))
+
+    @api.depends('line_ids.journal_id')
+    def _compute_journal_id(self):
+        for statement in self:
+            statement.journal_id = statement.line_ids.journal_id[:1]
+
+    @api.depends('is_valid_balance_end',)
+    def _compute_state(self):
+        for statement in self:
+            if statement.state == 'posted':
+                # we do not change the state of a posted statement automatically
+                continue
+            elif statement.is_valid_balance_end:
+                statement.state = 'complete'
+            else:
+                statement.state = 'open'
+
+    def _compute_attachment_ids(self):
+        for statement in self:
+            attachment_ids = self.env['ir.attachment'].search([
+                ('res_id', '=', statement.id),
+                ('res_model', '=', 'account.bank.statement')
+            ]).ids
+            message_attachment_ids = statement.mapped('message_ids.attachment_ids').ids  # from mail_thread
+            statement.attachment_ids = [Command.set(list(set(attachment_ids) - set(message_attachment_ids)))]
+
+    def _inverse_attachment_ids(self):
+        # remove the attachments deleted in the form
+        # the many2many_binary widget creates the attachments automatically, but it doesn't remove them when deleted
+        self.env['ir.attachment'].search([
+                ('res_id', 'in', self.ids),
+                ('res_model', '=', self._name),
+                ('id', 'not in', self.attachment_ids.ids)
+            ]).unlink()
```

Comment:
I don't understand the logic here. Could be nice to test in the interface to understand myself what you are trying to do with that but both demo data/views are broken :s

Reply:
The goal is to have attachments as both in a popup form view (without chatter) and a normal form view in chatter, synced.
The second view is deprecated and will be removed in future.

Reply:
Chatter is removed (with the old view) so it is not needed anymore